hamilton/node.py

```diff
@@ -370,3 +370,55 @@ def new_callable(**kwargs) -> Any:
             return __transform(self.callable(**kwargs), kwargs)
 
         return self.copy_with(callabl=new_callable, typ=__output_type)
+
+
+def matches_query(
+    tags: Dict[str, Union[str, List[str]]], query_dict: Dict[str, Optional[Union[str, List[str]]]]
+) -> bool:
+    """Check whether a set of node tags matches the query based on tags.
+
+    :param tags: the tags of the node.
+    :param query_dict: of tag to value. If value is None, we just check that the tag exists.
+    :return: True if we have tags that match all tag queries, False otherwise.
+    """
+    if not query_dict:
```

Comment:
Feels like `tag_filter={}` should trigger on everything? Or not be allowed, cause its pointless? No filter IMO means it matches. Also that means you can just remove this.

Reply:
done.